bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/PointType.java

```diff
@@ -78,9 +81,6 @@ public PointType(StringType latitude, StringType longitude, StringType altitude)
     }
 
     public PointType(String value) {
-        if (value == null) {
```

Comment:
same here, all fields are properly initialized to fulfil none-null-ness.
The question is: do we want the framework to throw exceptions (e.g. stop executing a rule when an empty point is given) or continue with the default data from the constructors?

Reply:
Another thing: In a DSL rule the annotations won't have any effect. So calling this constructor with `null` is still possible.

Reply:
See below... It is also possible.

Reply:
If null is provided `value.isEmpty()` will throw a NPE, so I don't see any reason to throw a IAE and use an additional check.